spec/requests/api_new/users_controller_request_spec.rb

```diff
@@ -3,90 +3,103 @@
+			it 'returns unauthorized when not logged in' do
+				get "/api_new/users/current"
+				expect(response).to have_http_status(:unauthorized)
+			end
 		end
-	end
 
-	context 'for a nonprofit admin' do 
+		context 'for a nonprofit admin' do 
 
-		let(:user) {create(:user_base, roles:[build(:role_base, :as_nonprofit_admin) ])}
+			let(:user) {create(:user_base, roles:[build(:role_base, :as_nonprofit_admin) ])}
 
+			subject(:body) { response.body}
+			before do
+				sign_in user
+				get "/api_new/users/current"
+			end
 
-		subject(:body) { response.body}
-		before do
-			sign_in user
-			get "/api_new/users/current"
-		end
+			it {
+				expect(response).to have_http_status(:success)
+			}
 
-		it {
-			expect(response).to have_http_status(:success)
-		}
-
-		it {
-			is_expected.to include_json(
-				object: 'user', 
-				is_super_admin: false,
-				roles: [
-					{
-						host: Nonprofit.first.to_houid
-					}
-			])
-		}
-			
-	end
+			it {
+				is_expected.to include_json(
+					object: 'user', 
+					is_super_admin: false,
+					roles: [
+						{
+							host: Nonprofit.first.to_houid
+						}
+				])
+			}
 
-	context 'for a nonprofit associate' do 
-		let(:user) {create(:user_as_nonprofit_associate)}
+		end
 
-		subject(:body) { response.body}
-		before do
-			sign_in user
-			get "/api_new/users/current"
+		context 'for a nonprofit associate' do 
+			let(:user) {create(:user_as_nonprofit_associate)}
+
+			subject(:body) { response.body}
+			before do
+				sign_in user
+				get "/api_new/users/current"
+			end
+
+			it {
+				expect(response).to have_http_status(:success)
+			}
+
+			it {
+				is_expected.to include_json(
+					object: 'user', 
+					is_super_admin: false,
+					roles: []
+				)
+			}
+				
 		end
 
-		it {
-			expect(response).to have_http_status(:success)
-		}
-
-		it {
-			is_expected.to include_json(
-				object: 'user', 
-				is_super_admin: false,
-				roles: []
-			)
-		}
-			
+		context "for super admin" do
+			let(:user) {create(:user_as_super_admin )}
+
+			subject(:body) { response.body}
+			before do
+				sign_in user
+				get "/api_new/users/current"
+			end
+
+			it {
+				expect(response).to have_http_status(:success)
+			}
+
+			it {
+				is_expected.to include_json(
+					object: 'user', 
+					is_super_admin: true,
+					roles: []
+				)
+			}
+		end
 	end
 
-	context "for super admin" do
-		let(:user) {create(:user_as_super_admin )}
+	context 'GET /api_new/users/:user_id/current_nonprofit_object_events' do
 
-		subject(:body) { response.body}
-		before do
-			sign_in user
-			get "/api_new/users/current"
-		end
-
-		it {
-			expect(response).to have_http_status(:success)
-		}
-
-		it {
-			is_expected.to include_json(
-				object: 'user', 
-				is_super_admin: true,
-				roles: []
-			)
-		}
-	end
-		
+		context 'for a nonprofit associate' do
+			let(:user) {create(:user_as_nonprofit_associate)}
 
+			before do
+				sign_in user
+				get "/api_new/users/#{user.id}/current_nonprofit_object_events"
```

Comment:
The route MUST not have the `user.id` in it. That's moving the original problem we were trying to fix.

Look at how `current` works in routes: we need this route to be: `/api_new/users/current_nonprofit_object_events`